packages/driver/src/cy/net-stubbing/events/index.ts

```diff
@@ -1,21 +1,21 @@
-import { Route, Interception } from '../types'
-import { NetEventFrames } from '@packages/net-stubbing/lib/types'
-import { onRequestReceived } from './request-received'
-import { onResponseReceived } from './response-received'
-import { onRequestComplete } from './request-complete'
+import { Route, Interception, StaticResponse, NetEvent } from '../types'
+import { onBeforeRequest } from './before-request'
+import { onResponse } from './response'
+import { onAfterResponse } from './after-response'
 import Bluebird from 'bluebird'
+import { getBackendStaticResponse } from '../static-response-utils'
 
-export type HandlerFn<Frame extends NetEventFrames.BaseHttp> = (Cypress: Cypress.Cypress, frame: Frame, opts: {
+export type HandlerFn<D> = (Cypress: Cypress.Cypress, frame: NetEvent.ToDriver.Event<D>, userHandler: (data: D) => void | Promise<void>, opts: {
   getRequest: (routeHandlerId: string, requestId: string) => Interception | undefined
   getRoute: (routeHandlerId: string) => Route | undefined
   emitNetEvent: (eventName: string, frame: any) => Promise<void>
-  failCurrentTest: (err: Error) => void
-}) => Promise<void> | void
+  sendStaticResponse: (requestId: string, staticResponse: StaticResponse) => void
+}) => Promise<D> | D
 
 const netEventHandlers: { [eventName: string]: HandlerFn<any> } = {
-  'http:request:received': onRequestReceived,
-  'http:response:received': onResponseReceived,
-  'http:request:complete': onRequestComplete,
+  'before:request': onBeforeRequest,
+  'response': onResponse,
+  'after:response': onAfterResponse,
```

Comment:
the next step after this PR is to add `request` and `before-response` events here, and expose `req.on` to the user